packages/node/src/indexer/api.service.ts

```diff
@@ -206,12 +206,16 @@ export class ApiService
         });
       },
     );
-
     return this;
   }
 
-  updateBlockFetching(): void {
-    const onlyEventHandlers = isOnlyEventHandlers(this.project);
+  async updateChainTypes(newChainTypes: unknown): Promise<void> {
+    const chainTypes = await updateChainTypesHasher(newChainTypes);
+    await this.connectionPoolService.updateChainTypes(chainTypes);
+  }
+
+  updateBlockFetching(project?: SubqueryProject): void {
```

Comment:
Theres no need to pass through the newChainTypes/project in these. `this.project` will be the updated project